File: payola/s2js/runtime/client/src/main/scala/s2js/runtime/client/scala/collection/SeqCompanion.scala
package s2js.runtime.client.scala.collection

import s2js.compiler.javascript

trait SeqCompanion[A >: Null <: Seq]
{
    def empty: Any

    @javascript("""
        var a = self.empty();
        a.setInternalJsArray(jsArray);
        return a;
    """)
    def fromJsArray(jsArray: Any): A = null

    // Just a hack to make the map function work.
    def canBuildFrom: Boolean = true
}

File: payola/web/server/public/javascripts/runtime/s2js/runtime/scala/collection/SeqCompanion.js
goog.provide('scala.collection.SeqCompanion');
scala.collection.SeqCompanion = function() {
var self = this;
};
scala.collection.SeqCompanion.prototype.fromJsArray = function(jsArray) {
var self = this;

        var a = self.empty();
        a.setInternalJsArray(jsArray);
        return a;
    };
scala.collection.SeqCompanion.prototype.__class__ = new s2js.Class('scala.collection.SeqCompanion', []);
